etna/analysis/eda_utils.py

```diff
@@ -189,30 +194,93 @@ def sample_acf_plot(
         number of random segments to plot
     lags:
         number of timeseries shifts for cross-correlation
+    partial:
+        plot autocorrelation or partial autocorrelation
+    columns_num:
+        number of columns in subplots
     segments:
         segments to plot
     figsize:
         size of the figure per subplot with one segment in inches
     """
     if segments is None:
-        segments = sorted(ts.segments)
+        exist_segments = sorted(ts.segments)
+        chosen_segments = np.random.choice(exist_segments, size=min(len(exist_segments), n_segments), replace=False)
+        segments = list(chosen_segments)
 
-    k = min(n_segments, len(segments))
-    columns_num = min(2, k)
-    rows_num = math.ceil(k / columns_num)
+    title = "Partial Autocorrelation" if partial else "Autocorrelation"
+
+    fig, ax = prepare_axes(num_plots=len(segments), columns_num=columns_num, figsize=figsize)
+    fig.suptitle(title, fontsize=16)
+
+    for i, name in enumerate(segments):
+        df_slice = ts.to_pandas()[name]["target"]
+        if partial:
+            # for partial autocorrelation remove NaN from the beginning and end of the series
+            indices_nan = np.argwhere(np.isnan(df_slice.values)).squeeze(1)
+            indices2remove_begin = []
+            current_begin = 0
+            for i in range(len(indices_nan)):
+                if indices_nan[i] == current_begin:
+                    indices2remove_begin.append(indices_nan[i])
+                    current_begin += 1
+
+            indices2remove_end = []
+
+            current_end = len(df_slice.values) - 1
+            for i in range(len(indices_nan) - 1, -1, -1):
+                if indices_nan[i] == current_end:
+                    indices2remove_end.append(indices_nan[i])
+                    current_end -= 1
+
+            candidates2delete = indices2remove_begin + indices2remove_end
+            if set(indices_nan) != set(candidates2delete):
+                raise ValueError("There is a NaN in the middle of the time series!")
```

Comment:
Errors should be moved to `Raises` block in a docstring. Look at docstring of `cross_corr_plot` for example.